src/VisualStudio/Core/Impl/CodeModel/InternalElements/CodeFunction.IMethodXML.cs

```diff
@@ -5,13 +5,23 @@
 using System.Threading;
 using Microsoft.CodeAnalysis.Internal.Log;
 using Microsoft.VisualStudio.LanguageServices.Implementation.CodeModel.Interop;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.CodeModel.InternalElements
 {
     public partial class CodeFunction
     {
         public string GetXML()
```

Comment:
:question: So what would a non-hacky solution look like?

Reply:
OK, so what I'm liking right now is the following:

1. Update `FileCodeModel.TryGetDocument` to include a call to `GetOpenDocumentInCurrentContextWithChanges`, or something equivalent
2. Ensure the source text used for the previous item accounts for the possibility of work still needing to be done

Reply:
Well, the real solution is to rewrite how I handle open document wireup to not have the RDT threading issues, so then it all goes away.

Yes, we could move this call out of GetXml and into somewhere else. I was choosing this since it's a code path taken by only a small number of designers so the impact would be minimal.

Reply:
@sharwell: I've moved the call to TryGetDocument.